### DISCUSSION TITLE
Could we use plug-in system not only on start but on tag `micro-app`  declaration level

### DISCUSSION THREAD

**Discussion by Relaxe111** [jd-opensource/micro-app#762] [Ideas]
I have an use case where i need to declare micro apps dynamically  on runtime and i need to do some manipulations with help of `modules` like we can do on start({ modules:{...}), but on <micro-app .../> level:
```
//main.ts how is done now
microApp.start({
  plugins: {
  
    modules: {
      'calendar': [{
        loader(code, url, options, info) {
          if (url === 'xxx.js') {
            code = code.replace('var abc =', 'window.abc =')
          }
          return code
        }
      }],
  
      }]
    }
  }
})
```
I would need to do like this:
```
// MfCalendar.ts 


function MfCalendar() {
    return <micro-app
        iframe
        //i need to do same thing as i can do with start plugins.modules['calendar']
        module: [{
        loader(code, url, options, info) {
          if (url === 'xxx.js') {
            code = code.replace('var abc =', 'window.abc =')
          }
          return code
        }
      }]
    />
}
```
How one can do that?

**Comment by bailicangdu** [comment]
micro-app is customElement, not component, property must be string

**Comment by Relaxe111** [comment]
Is possible somehow to redeclare/update  plugins.modules  after i called start() ?

**Comment by bailicangdu** [comment]
not impossible, the function needs to convert to a string, not very good

**Comment by Relaxe111** [comment]
I have situation where i need to create micro app at runtime. So because name is required to be unique i need to create name on run time. for example i need multiple instances of the `calendar` micro-app to ensure name is unique i set auto increment number at the end of the name lie `calendar_1`, `calendar_2` so on. I have as well an  custom loader() which i need to declare on start(). Problem is that when i call start i do not have unique name yet, even though i know that for all `calendar_*` apps which `loader` I have to use. I need to tell somehow, micro app about updated list of micro-app names with proper loaders, such that micro-app know how to handle resources.
Is there any recommendation how i would do that?

**Reply by bailicangdu** [reply]
this is currently difficult to achieve
